In [1]:
import sys
import os
sys.path.append(os.path.abspath('../scripts'))
import importlib
import compare_countries
%matplotlib inline
importlib.reload(compare_countries)  # Reload to ensure latest module

<module 'compare_countries' from 'c:\\Users\\Lidya\\Documents\\projects\\solar-challenge-week0\\scripts\\compare_countries.py'>

In [2]:
# Define file paths
file_paths = {
    'Benin': '../data/benin_clean.csv',
    'Sierra Leone': '../data/sierraleone_clean.csv',
    'Togo': '../data/togo_clean.csv'
}

# Load data
dataframes = compare_countries.load_data(file_paths)


In [3]:
# Generate plots and results
compare_countries.create_boxplots(dataframes)
summary = compare_countries.compute_summary_table(dataframes)
print(summary.to_string(index=False))

Metric  Benin_Mean  Benin_Median  Benin_Std  Sierra Leone_Mean  Sierra Leone_Median  Sierra Leone_Std  Togo_Mean  Togo_Median   Togo_Std
   GHI  513.411100        520.30 304.765569         445.471047                406.1        294.381895 454.081218        430.3 319.096010
   DNI  356.480524        352.25 279.731019         256.390219                166.3        263.173538 296.409415        233.8 283.534417
   DHI  247.397346        227.80 143.420836         251.600694                242.4        141.632581 228.166180        211.4 150.064955


In [4]:
# Summary Table
summary_table = compare_countries.compute_summary_table(dataframes)
print("Summary Table:")
print(summary_table.to_string(index=False))

Summary Table:
Metric  Benin_Mean  Benin_Median  Benin_Std  Sierra Leone_Mean  Sierra Leone_Median  Sierra Leone_Std  Togo_Mean  Togo_Median   Togo_Std
   GHI  513.411100        520.30 304.765569         445.471047                406.1        294.381895 454.081218        430.3 319.096010
   DNI  356.480524        352.25 279.731019         256.390219                166.3        263.173538 296.409415        233.8 283.534417
   DHI  247.397346        227.80 143.420836         251.600694                242.4        141.632581 228.166180        211.4 150.064955


## Perform Statistical Testing

In [5]:
# Cell 4: Statistical Testing
stats_results = compare_countries.perform_statistical_tests(dataframes, metric='GHI')
print("Statistical Test Results:")
print(f"ANOVA p-value: {stats_results['ANOVA_pvalue']:.4f}")
print(f"Kruskal-Wallis p-value: {stats_results['Kruskal_pvalue']:.4f}")

Statistical Test Results:
ANOVA p-value: 0.0000
Kruskal-Wallis p-value: 0.0000


## Key Observations
- Benin shows the highest median GHI (~520.3 W/m²), indicating the strongest solar potential among the three countries.
- Togo has the greatest GHI variability (std ~319.1), suggesting inconsistent solar conditions possibly due to seasonal changes.
- Sierra Leone exhibits a more stable GHI profile (median ~406.1 W/m², std ~294.4), making it suitable for consistent energy generation.

## Visual Summary

In [6]:
# Cell 6: Bar Chart
import matplotlib.pyplot as plt
%matplotlib inline

compare_countries.create_bar_chart(dataframes, metric='GHI')
plt.show()

In [12]:
compare_countries.generate_observations(dataframes)

['Benin shows the highest median GHI (~520.3 W/m²), indicating strong solar potential.',
 'Togo has the greatest GHI variability (std ~319.1), suggesting inconsistent solar conditions.',
 'Sierra Leone exhibits a more stable GHI profile with a median of ~406.1 W/m², suitable for consistent energy generation.']